app/helpers/application_helper/button/auth_key_pair_cloud_delete.rb

```diff
@@ -0,0 +1,9 @@
+class ApplicationHelper::Button::AuthKeyPairCloudDelete < ApplicationHelper::Button::Basic
+  needs :@record
+
+  def disabled?
+    return true if @record.auth_key.present?
+    @error_message = _('Deletion is unavailable for this keypair.')
+    @error_message.present?
+  end
```

Comment:
Well this method will return `true` in any case.

Reply:
Yeah, this method will never return false.

This also feels like the wrong place to make this determination.  I would think whether or not something is "deletable" is up to each provider, right?

Reply:
Yeah I see that now :( I was trying to keep the same as https://github.com/ManageIQ/manageiq-ui-classic/blob/master/app/helpers/application_helper/button/auth_key_pair_cloud_download.rb Thanks :)